safe-transaction-client/src/lib.rs

```diff
@@ -0,0 +1,343 @@
+    Error<S::Error>: From<S::Error>,
+{
+    let mut signature = signer.sign_message(safe_tx_hash).await?;
+
+    // signature.recover expects signature.v ∈ {27, 28}
+    if signature.v < 2 {
+        signature.v += 27;
+    }
+    assert!(
+        [27, 28].contains(&signature.v),
+        "BUG: `Signer::sign_message` produced an invalid `v` field."
+    );
+    assert_eq!(
+        Some(signer.address()),
+        signature.recover(&safe_tx_hash[..]).ok(),
+        "BUG: `Signature::recover` failed to recover our signing \
+            address... did `Signer::sign_message` fail to add the prefix?"
+    );
+    signature.v += 4;
+
+    Ok(SignedSafeTxHash {
+        safe_tx_hash,
+        signature,
+    })
+}
+
+#[test]
+#[cfg(feature = "run-test-against-api")]
+fn check() {
+    futures_executor::block_on(async {
+        use rand_chacha::rand_core::SeedableRng;
+
+        // super secure RNG for tests
+        let mut fake_rng =
+            rand_chacha::ChaCha20Rng::from_seed(*b"0123456789abcdef0123456789abcdef");
+        let wallet = LocalWallet::new(&mut fake_rng).with_chain_id(4u64);
+
+        // [x] propose a new transaction
+
+        let value = U256::from(0u64);
+        let _data_unanchor =
+            hex::decode("cbd566010000000000000000000000000000000000000000000000000000000000000000")
+                .unwrap();
+        let data = hex::decode("68f1fbf80000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000000").unwrap();
+
+        let client = Client::new("https://safe-transaction.rinkeby.gnosis.io/api");
+        let safe = client.with_safe(
+            Address::from_str("0xb535CEd5f003e00CfF2424892D4885b139019F1d")
+                .expect("Invalid rinkeby address"),
+        );
+        let org_rinkeby = Address::from_str("0xaFb752f961CEF7FdfB9d2925120D23Aa9B4ed7Ae")
+            .expect("Invalid org address");
+        let safe_tx = safe
+            .create_transaction(org_rinkeby, value, Bytes::from(data), Operation::Call)
+            .expect("Transaction creation error?");
+
+        // Optionally update any of safe_tx' fields. The nonce is dynamically
+        // set respective to the safe state.
+        let signed_safe_tx = dbg!(safe_tx)
+            .sign(&wallet)
+            .await
+            .expect("Invalid signature");
+        dbg!(safe.propose(signed_safe_tx)).expect("Failed to propose transaction");
+
+        // [x] add confirmation to existing transaction
+
+        let safe_tx_hash =
+            TxHash::from_str("0xec812245f4c8908551914922bb1b093b3aa8c02e284c5e8c54fb8170673716cd")
+                .expect("Invalid TxHash");
+
+        let _ = dbg!(safe.add_to_confirmations(
+            sign_tx_hash(&wallet, safe_tx_hash)
+                .await
+                .expect("Invalid signature")
+        ));
```

Comment:
Can we remove the `dbg!()`s?

Reply:
Removed & error handling slightly adjusted for the confirm case -- to actually panic the test if the confirmation failed (but not because the transaction was already executed).